microservices-connector/internal/controller/gmconnector_controller.go

```diff
@@ -362,23 +370,17 @@ func (r *GMConnectorReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 
 	for node, router := range graph.Spec.Nodes {
 		for i, step := range router.Steps {
-			fmt.Println("reconcil service for node:", step.StepName)
+			fmt.Println("reconcile resource for node:", step.StepName)
 
 			if step.Executor.ExternalService == "" {
 				ns := step.Executor.InternalService.NameSpace
 				svcName := step.Executor.InternalService.ServiceName
-				fmt.Println("trying to find internal service [", svcName, "] in namespace ", ns)
+				fmt.Println("trying to internal service [", svcName, "] in namespace ", ns)
 
 				service := &corev1.Service{}
-				err := r.Client.Get(ctx, types.NamespacedName{Namespace: ns, Name: svcName}, service)
-				if err == nil {
-					fmt.Println("success to get service ", svcName)
-				} else {
-					fmt.Println("service [", svcName, "] is needed, start to deploy ...")
-					err := createService(step.StepName, ns, svcName, &step.Executor.InternalService.Config, service)
-					if err != nil {
-						return reconcile.Result{Requeue: true}, errors.Wrapf(err, "Failed to create service for %s", svcName)
-					}
+				err := reconcileResource(step.StepName, ns, svcName, &step.Executor.InternalService.Config, service)
+				if err != nil {
+					return reconcile.Result{Requeue: true}, errors.Wrapf(err, "Failed to create service for %s", svcName)
```

Comment:
do you want the verb "create" here or "reconcile/patch/update" in "Failed to .. service"